### PR TITLE
Fix all pep8 and pyflakes failures, add helper run_tests script

### PR DESCRIPTION
**Pull request by ahayzen**
Understand if you don't want this, not everyone wants 79 char line lengths :-)

Before
```
$ pep8 .
./src/checker.py:27:1: E302 expected 2 blank lines, found 1
./src/checker.py:30:1: E302 expected 2 blank lines, found 1
./src/checker.py:37:80: E501 line too long (81 > 79 characters)
./src/checker.py:38:80: E501 line too long (82 > 79 characters)
./src/checker.py:41:80: E501 line too long (84 > 79 characters)
./src/checker.py:42:80: E501 line too long (81 > 79 characters)
./src/checker.py:43:80: E501 line too long (87 > 79 characters)
./src/checker.py:48:80: E501 line too long (82 > 79 characters)
./src/checker.py:49:80: E501 line too long (89 > 79 characters)
./src/checker.py:54:75: E502 the backslash is redundant between brackets
./src/checker.py:99:18: E201 whitespace after '{'
./src/checker.py:122:25: E703 statement ends with a semicolon
./src/lib/externaldata.py:27:1: E302 expected 2 blank lines, found 1
./src/lib/externaldata.py:40:23: E261 at least two spaces before inline comment
./src/lib/externaldata.py:41:24: E261 at least two spaces before inline comment
./src/lib/externaldata.py:70:80: E501 line too long (81 > 79 characters)
./src/lib/externaldata.py:91:1: E302 expected 2 blank lines, found 1
./src/lib/externaldata.py:96:1: E302 expected 2 blank lines, found 1
./src/lib/externaldata.py:102:80: E501 line too long (82 > 79 characters)
./src/lib/utils.py:23:1: E302 expected 2 blank lines, found 1
./src/lib/utils.py:31:1: E302 expected 2 blank lines, found 1
./src/lib/utils.py:36:1: E302 expected 2 blank lines, found 1
./src/checkers/debianrepochecker.py:45:1: E302 expected 2 blank lines, found 1
./src/checkers/debianrepochecker.py:74:80: E501 line too long (85 > 79 characters)
./src/checkers/debianrepochecker.py:76:1: E302 expected 2 blank lines, found 1
./src/checkers/debianrepochecker.py:88:80: E501 line too long (90 > 79 characters)
./src/checkers/debianrepochecker.py:158:51: E128 continuation line under-indented for visual indent
./src/checkers/debianrepochecker.py:179:1: E305 expected 2 blank lines after class or function definition, found 1
./src/checkers/rotatingurlchecker.py:37:1: E302 expected 2 blank lines, found 1
./src/checkers/rotatingurlchecker.py:46:80: E501 line too long (91 > 79 characters)
./src/checkers/rotatingurlchecker.py:69:1: E305 expected 2 blank lines after class or function definition, found 1
./src/checkers/urlchecker.py:27:1: E302 expected 2 blank lines, found 1
./src/checkers/urlchecker.py:32:9: E722 do not use bare except'
./src/checkers/urlchecker.py:37:1: E305 expected 2 blank lines after class or function definition, found 1
./tests/test_checker.py:29:1: E402 module level import not at top of file
./tests/test_checker.py:30:1: E402 module level import not at top of file
./tests/test_checker.py:32:80: E501 line too long (80 > 79 characters)
./tests/test_checker.py:39:1: E302 expected 2 blank lines, found 1
./tests/test_checker.py:42:80: E501 line too long (87 > 79 characters)
./tests/test_checker.py:45:1: E302 expected 2 blank lines, found 1
./tests/test_checker.py:87:80: E501 line too long (84 > 79 characters)
./tests/test_checker.py:90:80: E501 line too long (81 > 79 characters)
./tests/test_checker.py:96:1: E305 expected 2 blank lines after class or function definition, found 1

$ pyflakes3 .
./src/lib/externaldata.py:75: undefined name '__class__'
./src/lib/externaldata.py:76: undefined name '__class__'
```

After

```
$ pep8 .
$ pyflakes3 .
```

I also added a `run_tests.sh` script to set the PYTHONPATH instead of using the `sys` module, maybe there should be a `run_linters.sh` or something if you accept this for pep8 + pyflakes3 ?

### CODE REVIEW

File: src/checker.py [hunk 1]
```diff
@@ -24,34 +24,43 @@
...
+                                    manifest_file.read())
+            self._json_data = json.loads(clean_manifest,
+                                         object_pairs_hook=OrderedDict)
 
         self._collect_external_data()
 
     def _collect_external_data(self):
-        self._external_data = self._get_module_data_from_json(self._json_data) + \
-                              self._get_finish_args_extra_data_from_json(self._json_data)
+        self._external_data = (
```
Comment (joaquimrocha) [inline src/checker.py]: Is this expected by pep8? I really don't love this change.

File: src/checker.py [hunk 2]
```diff
@@ -119,7 +128,7 @@ def check(self, filter_type=None):
         for data in self._external_data:
             # Ignore if the type is not the one we care about
             if filter_type is not None and filter_type != data.type:
-                continue;
```
Comment (joaquimrocha) [inline src/checker.py:122]: Oops.

File: src/lib/externaldata.py [hunk 3]
```diff
@@ -53,27 +54,28 @@ def __init__(self, data_type, filename, url, checksum, size=-1, arches=[],
...
-               '  Arches:  {arches}\n' \
-               '  Checker: {checker_data}'.format(state=self.state.name,
-                                                  filename=self.filename,
-                                                  type=self.type.name,
-                                                  url=self.url,
-                                                  checksum=self.checksum,
-                                                  size=self.size,
-                                                  arches=self.arches,
-                                                  checker_data=self.checker_data)
+        info = (
```
Comment (joaquimrocha) [inline src/lib/externaldata.py]: Again, I don't love this change.

File: src/lib/externaldata.py [hunk 4]
```diff
@@ -53,27 +54,28 @@ def __init__(self, data_type, filename, url, checksum, size=-1, arches=[],
...
+            )
+        )
+
         return info
 
     def to_json(self):
         json_data = OrderedDict()
-        json_data['type'] = __class__._TYPES_MANIFEST_MAP[self.type]
-        json_data[__class__._TYPES_MANIFEST_MAP[self.type]] = self.filename
+        json_data['type'] = self._TYPES_MANIFEST_MAP[self.type]
```
Comment (joaquimrocha) [inline src/lib/externaldata.py]: Did you run this in Python 3? I just wanted to use the class itself here.

File: tests/test_checker.py [hunk 5]
```diff
@@ -84,14 +83,17 @@ def test_check(self):
         file_ext_data = self.checker.get_external_data(ExternalData.Type.FILE)
         self.assertEqual(len(file_ext_data), NUM_FILE_IN_MANIFEST)
 
-        archive_ext_data = self.checker.get_external_data(ExternalData.Type.ARCHIVE)
+        archive_ext_data = self.checker.get_external_data(
+            ExternalData.Type.ARCHIVE)
```
Comment (joaquimrocha) [inline tests/test_checker.py]: I don't like that this is to the left of the `=` sign. Does it fit if you bring the `self.checker...` to the line below?

**Review by joaquimrocha — CHANGES_REQUESTED** [review]
I like the general idea of having a lint in order to make the code coherent. For Python I think usually the code already turns out very consistent anyway, but I don't oppose to using a lint here either. It does seem a bit too harsh to enforce 80 chars in today's displays, but it's the weird formatting to accommodate the limit that I dislike. Please check my comments, and I can accept this if in the end the formatting is not completely crazy.

File: src/lib/externaldata.py [hunk 6]
```diff
@@ -88,18 +90,21 @@ def to_json(self):
...
+
 class CheckerRegistry:
 
     _checkers = []
 
     @staticmethod
     def load(checkers_folder):
-        for _unused, modname, _unused in pkgutil.walk_packages([checkers_folder]):
+        for _unused, modname, _unused in pkgutil.walk_packages(
+                [checkers_folder]):
```
Comment (joaquimrocha) [inline src/lib/externaldata.py]: I don't like opening parenthesis without args in that line. I'd prefer to break after the `in`.

**Comment by ahayzen** [comment]
@joaquimrocha Thanks for the reviews! :-) As you agree with most of the changes, other than the line length ones. How would you feel about setting the line length to, eg 100 or 120 chars ? (As this will likely solve most of the wrapping issues).

pep8 has an option to override this (as it is a debatable option), eg we could run the linting with `pep8 --max-line-length=100` (or any other length). And then we could add the linter's with their options to either a separate script or built into the run_tests script.

I'll try and go through the rest of your requests today or tomorrow :-)

**Comment by ahayzen** [comment]
Hmm seems merge conflict resolutions have either caused the diff to be wrong, or the diff is confused. I'll investigate tomorrow. Don't merge for now.

**Comment by ahayzen** [comment]
OK, so turns out that github is showing the diff of all the commits, instead of the diff against master - maybe i'm holding it wrong ;-)  The diff can be seen here https://github.com/joaquimrocha/flatpak-external-data-checker/compare/master...ahayzen:pep8-pyflakes

@joaquimrocha could you re-review now that max line length is 100 ?

**Comment by joaquimrocha** [comment]
@ahayzen I was trying to review this today but it's no longer applying correctly to master (I know it's my fault since it took me a while to do it). However, I noticed that you call `pep8` which seems to have changed into `pycodestyle` and apparently `pyflakes3` doesn't exist in Fedora (it's `pyflakes-3`).

It seems like `flake8` is a much more powerful alternative (it uses the mentioned ones as plugins IIRC), so if you want use that instead, please open a new PR.

**Pull request closed without merging**